tests/PHPUnit/Unit/PeriodTest.php

```diff
@@ -123,4 +123,145 @@ public function getInvalidDateFormats()
             array(3434),
         );
     }
+
+    /**
+     * @dataProvider getTestDataForGetAllParentPeriods
+     */
+    public function test_getAllParentPeriods_ReturnsCorrectContainingPeriods($periodType, $dateRange, $expectedParentPeriodRanges)
+    {
+        $period = Period\Factory::build($periodType, $dateRange);
+
+        $parentPeriods = $period->getAllParentPeriods();
+        $parentPeriods = $this->getPeriodInfoForAssert($parentPeriods);
+
+        $this->assertEquals($expectedParentPeriodRanges, $parentPeriods);
+    }
+
+    public function getTestDataForGetAllParentPeriods()
+    {
+        return array(
+            array(
+                'day',
+                '2015-01-01',
+                array(
+                    array('week', '2014-12-29,2015-01-04'),
+                    array('month', '2014-12-01,2014-12-31'),
```

Comment:
The parent periods for a day are week, month, year. The week for 2015-01-01 is 2014-12-18,2015-01-03. There are two months touched by that week period.

I think I can remove this method though, there's a less complicated approach.